src/wallet/rpcwallet.cpp

```diff
@@ -1133,6 +1133,9 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
     if (params.size() > 0)
         nMinDepth = params[0].get_int();
 
+    bool fFilterAddress = !fByAccounts && params.size() > 3;
+    const CBitcoinAddress filterAddress = fFilterAddress ?  CBitcoinAddress(params[3].get_str()) : CBitcoinAddress{};
```

Comment:
nit: I think it's clearer to move these two lines below the `if(params.size() > 2)` block, so the params are being tested in order. I also have a slight preference to change this to a `if(params.size() > 3)` block to match the other tests.